tools/c7n_azure/c7n_azure/query.py

```diff
@@ -85,24 +88,38 @@ def filter(self, resource_manager, **params):
 
         enum_op, list_op, extra_args = m.enum_spec
 
-        parent_type, annotate_parent = m.parent_spec
-        parents = self.manager.get_resource_manager(parent_type)
+        parents = self.manager.get_resource_manager(m.parent_manager_name)
 
         # Have to query separately for each parent's children.
         results = []
         for parent in parents.resources():
             if extra_args:
-                params.update({key: parent[extra_args[key]] for key in extra_args.keys()})
```

Comment:
Can we add a few comments here to just specify the intention and assumptions, it will help with future refactoring?